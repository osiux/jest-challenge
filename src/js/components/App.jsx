import React from "react";

import Counter from "./Counter";
import Controls from "./Controls";

class App extends React.Component {
    constructor() {
        super();

        this.state = {
            counter: 0
        };
    }

    _increaseCounter = () =>
        this.setState(prevState => {
            return {
                counter: prevState.counter + 1
            };
        });

    _decreaseCounter = () =>
        this.setState(prevState => {
            return {
                counter: prevState.counter - 1
            };
        });

    _resetCounter = () =>
        this.setState({
            counter: 0
        });

    render() {
        const { counter } = this.state;

        return (
            <div className="container">
                <div className="row justify-content-center">
                    <div className="col-sm-12 col-md-6">
                        <h1 className="text-center">Counter</h1>

                        <div className="card text-center">
                            <Counter counter={counter} />

                            <Controls
                                onDecrease={this._decreaseCounter}
                                onReset={this._resetCounter}
                                onIncrease={this._increaseCounter}
                            />
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default App;
