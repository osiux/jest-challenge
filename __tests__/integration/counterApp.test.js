import React from 'react';
import { mount } from 'enzyme';

import App from '../../src/js/components/App';


describe('Counter App', () => {
    it('matches snapshot', () => {
        const component = mount(<App />);

        expect(component).toMatchSnapshot();
    });

    it('has 0 as counter initial value', () => {
        const component = mount(<App />);

        expect(component.state('counter')).toBe(0);
        expect(component.find('p.counter').text()).toBe('0');
    });

    it('decrease button decreases the counter value', () => {
        const component = mount(<App />);

        expect(component.state('counter')).toBe(0);

        const decreaseButton = component.find('button').first();
        decreaseButton.simulate('click');

        expect(component.state('counter')).toBe(-1);
        expect(component.find('p.counter').text()).toBe('-1');
    });

    it('increase button increases the counter value', () => {
        const component = mount(<App />);

        expect(component.state('counter')).toBe(0);

        const increaseButton = component.find('button').last();
        increaseButton.simulate('click');

        expect(component.state('counter')).toBe(1);
        expect(component.find('p.counter').text()).toBe('1');
    });

    it('reset button resets the counter value to 0', () => {
        const component = mount(<App />);

        component.setState({ counter: 10 });

        expect(component.state('counter')).toBe(10);

        const resetButton = component.find('button').at(1);
        resetButton.simulate('click');

        expect(component.state('counter')).toBe(0);
        expect(component.find('p.counter').text()).toBe('0');
    });
});