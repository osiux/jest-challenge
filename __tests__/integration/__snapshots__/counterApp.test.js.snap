// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter App matches snapshot 1`] = `
<App>
  <div
    className="container"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-sm-12 col-md-6"
      >
        <h1
          className="text-center"
        >
          Counter
        </h1>
        <div
          className="card text-center"
        >
          <Counter
            counter={0}
          >
            <div
              className="card-body"
            >
              <p
                className="counter"
                style={
                  Object {
                    "fontSize": "500%",
                  }
                }
              >
                0
              </p>
            </div>
          </Counter>
          <Controls
            onDecrease={[Function]}
            onIncrease={[Function]}
            onReset={[Function]}
          >
            <div
              className="card-body "
            >
              <div
                className="btn-group btn-group-justified"
              >
                <Button
                  context="primary"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-primary"
                    onClick={[Function]}
                  >
                    -
                  </button>
                </Button>
                <Button
                  context="danger"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-danger"
                    onClick={[Function]}
                  >
                    Reset
                  </button>
                </Button>
                <Button
                  context="primary"
                  onClick={[Function]}
                >
                  <button
                    className="btn btn-primary"
                    onClick={[Function]}
                  >
                    +
                  </button>
                </Button>
              </div>
            </div>
          </Controls>
        </div>
      </div>
    </div>
  </div>
</App>
`;
